autorest/utility_test.go

```diff
@@ -234,6 +234,29 @@ func TestStringWithValidString(t *testing.T) {
 	}
 }
 
+func TestStringWithStringSlice(t *testing.T) {
+	s := []string{"string1", "string2"}
+	if String(s, ",") != "string1,string2" {
+		t.Fatal("autorest: String method failed to convert string slice to string")
+	}
+}
+
+type TestEnumType string
```

Comment:
[nit] I'd just declare this type inside of the tests that use it as to not pollute the namespace.